libs/api/domains/education/src/lib/education.service.ts

```diff
@@ -53,11 +55,26 @@ export class EducationService {
       nationalId,
       licenseId,
     )
-
-    return this.s3Service.uploadFileFromStream(responseStream, {
-      fileName: uuid(),
-      bucket: this.config.fileDownloadBucket,
-    })
+    try {
+      const fileLocation = await this.s3Service.uploadFile(
+        responseStream.body,
+        { bucket: this.config.fileDownloadBucket, key: uuid() },
+        {
+          ContentType:
+            responseStream.headers?.get('content-type') || 'application/pdf',
+        },
+      )
+
+      // Presigned URL expires in 65 seconds to allow for download initiation
+      const PRESIGNED_URL_EXPIRY = 65
+      return await this.s3Service.getPresignedUrl(
+        fileLocation,
+        PRESIGNED_URL_EXPIRY,
+      )
+    } catch (error) {
+      this.logger.error(`Failed to process PDF license: ${licenseId}`, error)
```

Comment:
One comment, put the `licenseId` as a property instead of concatenating it to the string. Makes it easier to filter in datadog

```suggestion
      this.logger.error('Failed to process PDF license', {error, licenseId})
```

Reply:
good point, fixed.